File: tests/wasm2lang_24_simd_dense.harness.cs
// Dense SIMD128 module (C#).
//
// Enumerates the module's own exports rather than listing them; see the .mjs
// harness for why.  StringComparer.Ordinal, not the culture-aware default: the
// other two harnesses sort by code unit, and a culture-aware sort would order
// the same ASCII names differently and turn every line into a diff.
public static class W2lHarness {
  public static void Run() {
    var foreign = new System.Collections.Generic.Dictionary<string, object>();
    var memBuffer = WasmMemBuffer.memBuffer();
    var mod = new WasmModule(foreign, memBuffer);
    var exports = new System.Collections.Generic.List<System.Reflection.MethodInfo>();
    foreach (var m in typeof(WasmModule).GetMethods(
                 System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.NonPublic |
                 System.Reflection.BindingFlags.Instance)) {
      if (m.GetParameters().Length == 0 && m.ReturnType == typeof(int) && m.Name.StartsWith("t_")) exports.Add(m);
    }
    exports.Sort((a, b) => System.StringComparer.Ordinal.Compare(a.Name, b.Name));
    System.Console.WriteLine("exports=" + exports.Count);
    foreach (var m in exports) System.Console.WriteLine(m.Name + "=" + m.Invoke(mod, null));
    W2l.DumpCRC(memBuffer);
  }
}

File: tests/wasm2lang_24_simd_dense.harness.java
// Dense SIMD128 module (Java).
//
// Enumerates the module's own exports rather than listing them; see the .mjs
// harness for why.  getDeclaredMethods, not getMethods: an exported wasm
// function is emitted package-private on this backend, so the public-only view
// would find nothing and the test would pass while calling zero functions.
// The printed count is the guard against exactly that.
{
    java.util.Map<String, Object> foreign = new java.util.LinkedHashMap<>();
    WasmModule mod = new WasmModule(foreign, memBuffer);
    java.util.List<java.lang.reflect.Method> exports = new java.util.ArrayList<>();
    for (java.lang.reflect.Method m : WasmModule.class.getDeclaredMethods()) {
        if (m.getParameterCount() == 0 && m.getReturnType() == int.class && m.getName().startsWith("t_")) {
            exports.add(m);
        }
    }
    exports.sort(java.util.Comparator.comparing(java.lang.reflect.Method::getName));
    System.out.println("exports=" + exports.size());
    for (java.lang.reflect.Method m : exports) {
        m.setAccessible(true);
        System.out.println(m.getName() + "=" + m.invoke(mod));
    }
    w2lDumpCRC(memBuffer);
}

/exit
